src/EventStore.Core/Services/Storage/StorageScavenger.cs

```diff
@@ -52,6 +56,18 @@ public void Handle(SystemMessage.StateChangeMessage message)
             }
         }
 
+        public void Handle(ClientMessage.MergeIndexes message)
```

Comment:
As mentioned [here](https://github.com/EventStore/EventStore/pull/1786/files#r244915589), it would be better to move this to `IndexCommiterService.cs` instead of `StorageScavenger.cs`